core/src/main/java/net/kyori/moonshine/Moonshine.java

```diff
@@ -0,0 +1,205 @@
+//
+// moonshine - A localisation library for Java.
+// Copyright (C) 2021 Mariell Hoversholm
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+package net.kyori.moonshine;
+
+import io.leangen.geantyref.GenericTypeReflector;
+import io.leangen.geantyref.TypeToken;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NavigableSet;
+import javax.annotation.concurrent.ThreadSafe;
```

Comment:
Would prefer to get rid of usage of jsr305, and move to JetBrains annotations. Not all are available, but it should be enough?

Reply:
Does JetBrains annotations provide `@ThreadSafe` and `@NotThreadSafe`?

Reply:
No. But we cannot continue using jsr305, either.

Reply:
Oh, is it scheduled for removal or something? Haven't heard of it in that case 🤔

Reply:
See https://github.com/google/guava/issues/2960 for context.

Reply:
I suppose we can replace all JSR-305 annotations with JetBrains/CheckerFramework/javadocs, then.